libtransmission/verify.cc

```diff
@@ -227,3 +282,89 @@ int tr_verify_worker::Node::compare(Node const& that) const noexcept
 
     return 0;
 }
+
+class SynchronousVerifyMediator final : public tr_verify_worker::Mediator
+{
+public:
+    explicit SynchronousVerifyMediator(
+        tr_torrent_metainfo const* metainfo,
+        std::string_view const* data_dir,
+        std::function<void(tr_file_index_t, bool, std::string)> file_status_cb)
+        : metainfo_{ metainfo }
+        , data_dir_{ data_dir }
+        , file_status_cb_{ std::move(file_status_cb) }
+    {
+    }
+
+    ~SynchronousVerifyMediator() override = default;
+
+    SynchronousVerifyMediator(SynchronousVerifyMediator const&) = delete;
+    SynchronousVerifyMediator(SynchronousVerifyMediator&&) = delete;
+    SynchronousVerifyMediator& operator=(SynchronousVerifyMediator const&) = delete;
+    SynchronousVerifyMediator& operator=(SynchronousVerifyMediator&&) = delete;
+
+    [[nodiscard]] tr_torrent_metainfo const& metainfo() const override
+    {
+        return *metainfo_;
+    }
+
+    [[nodiscard]] std::optional<std::string> find_file(tr_file_index_t file_index) const override
+    {
+        if (auto const found = metainfo_->files().find(file_index, data_dir_, 1); found)
+        {
+            return std::string{ found->filename().sv() };
+        }
+
+        return {};
+    }
+
+    void on_verify_queued() override
+    {
+    }
+
+    void on_verify_started() override
+    {
+        all_valid = true;
+    }
+
+    void on_piece_checked(tr_piece_index_t /*piece*/, bool /*has_piece*/) override
+    {
+    }
+
+    void on_verify_done(bool /*aborted*/) override
+    {
+    }
+
+    void on_file_ok(tr_file_index_t index) override
+    {
+        file_status_cb_(index, true, "");
+    }
+
+    void on_file_error(tr_file_index_t index, std::string error) override
+    {
+        file_status_cb_(index, false, error);
+        all_valid = false;
+    }
+
+    [[nodiscard]] bool torrent_is_valid() const
+    {
+        return all_valid;
+    }
+
+private:
+    tr_torrent_metainfo const* metainfo_;
+    std::string_view const* data_dir_;
+    std::function<void(tr_file_index_t, bool, std::string)> file_status_cb_;
+    bool all_valid{ false };
+};
+
+bool tr_torrentSynchronousVerify(
+    tr_torrent_metainfo const* metainfo,
+    std::string_view const* data_dir,
+    std::function<void(tr_file_index_t, bool, std::string)> file_status_cb)
+{
+    std::atomic<bool> const& abort_flag = false;
+    SynchronousVerifyMediator mediator(metainfo, data_dir, std::move(file_status_cb));
+    tr_verify_worker::verify_torrent(mediator, abort_flag, 0ms);
```

Comment:
What is the reason for removing the hard-coded 100ms break here?

Judging from this comment, it sounds to me sleeping every once in a while is always a good idea:

https://github.com/transmission/transmission/blob/0f321b850342646c8ca3fd5e54a75220e34865cf/libtransmission/verify.cc#L145-L146